dali/kernels/imgproc/paste/paste_gpu.h

```diff
@@ -0,0 +1,293 @@
+          }
+          patch.patch_start[1] *= channels;
+          patch.patch_end[1] *= channels;
+          patch.in_pitch[1] *= channels;
+          patch.in_anchor[1] *= channels;
+
+          // Now remove regions that end here
+          for (int i : ending[x + 1]) {
+            open_pastes.erase(i);
+          }
+        }
+        // And remove start/events for regions whose y ends here
+        for (auto &i : y_ending[y + 1]) {
+          starting[std::get<1>(i)].erase(std::get<0>(i));
+          ending[std::get<2>(i)].erase(std::get<0>(i));
+        }
+      }
+  }
+}
+
+
+
+template <class OutputType, class InputType, int ndims>
+__global__ void
+PasteKernel(const SampleDescriptorGPU<OutputType, InputType, ndims> *samples,
+                  const PatchDesc<InputType, ndims> *patchs,
+                  const BlockDesc<ndims> *blocks) {
+  static_assert(ndims == 2, "Function requires 2 dimensions in the input");
+  const auto &block = blocks[blockIdx.x];
+  const auto &sample = samples[block.sample_idx];
+  const PatchDesc<InputType, ndims> *my_patchs = patchs + sample.patch_start_idx;
+
+
+  auto *__restrict__ out = sample.out;
+
+  int patch_y = 0;
+  int min_patch_x = 0;
+  while (threadIdx.x + block.start.x >= my_patchs[min_patch_x].patch_end[1]) min_patch_x++;
+
+  for (int y = threadIdx.y + block.start.y; y < block.end.y; y += blockDim.y) {
+    while (y >= my_patchs[patch_y * sample.patch_counts[1]].patch_end[0]) patch_y++;
+    int patch_x = min_patch_x;
+    for (int x = threadIdx.x + block.start.x; x < block.end.x; x += blockDim.x) {
+      while (x >= my_patchs[patch_y * sample.patch_counts[1] + patch_x].patch_end[1]) patch_x++;
+      const PatchDesc<InputType, ndims> *patch =
+              &my_patchs[patch_y * sample.patch_counts[1] + patch_x];
+      if (patch->in == nullptr) {
+          out[y * sample.out_pitch[1] + x] = 0;
+      } else {
+        out[y * sample.out_pitch[1] + x] = ConvertSat<OutputType>(
+                patch->in[(y - patch->patch_start[0] + patch->in_anchor[0]) * patch->in_pitch[1]
+                         + (x - patch->patch_start[1] + patch->in_anchor[1])]);
+      }
+    }
+  }
+}
+
+}  // namespace paste
+
+template <typename OutputType, typename InputType, int ndims>
+class PasteGPU {
+ private:
+  static constexpr size_t spatial_dims = ndims - 1;
+  using BlockDesc = kernels::BlockDesc<spatial_dims>;
+  using SampleDesc = paste::SampleDescriptorGPU<OutputType, InputType, spatial_dims>;
+  using PatchDesc = paste::PatchDesc<InputType, spatial_dims>;
+
+  std::vector<SampleDesc> sample_descriptors_;
+  std::vector<PatchDesc> patch_descriptors_;
+
+ public:
+  BlockSetup<spatial_dims, -1 /* No channel dimension, only spatial */> block_setup_;
+
+  KernelRequirements Setup(
+      KernelContext &context,
+      const InListGPU<InputType, ndims> &in,
+      span<paste::MultiPasteSampleInput<spatial_dims>> samples,
+      TensorListShape<ndims> out_shape) {
```

Comment:
It's a relatively heavy object.
```suggestion
      const TensorListShape<ndims> &out_shape) {
```